library/alloc/src/vec/mod.rs

```diff
@@ -1925,6 +1929,7 @@ impl<T, A: Allocator> Vec<T, A> {
             let end = self.as_mut_ptr().add(self.len);
             ptr::write(end, value);
             self.len += 1;
```

Comment:
@c410-f3r

Just as a note: LLVM's constraint elimination can work better when this is written as:

```rust
// rustc emits: `add nuw`, so the pass can assume that the instruction does not overflow
self.len = unsafe { self.len.unchecked_add(1) };
```

This reduces some calls to `reserve_for_push` but does not eliminate all of them. I haven't really had the time to dig in further to find out...